### PR TITLE
Fix: Nested tabs styling issues regarding icons [ED-8843]

### PR DESCRIPTION
**Pull request by davseve**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [x] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: modules/nested-tabs/assets/scss/frontend.scss [hunk 1]
```diff
@@ -181,6 +166,26 @@
 					}
 				}
 			}
+
+			&:hover {
+
+			  	&,
+				&.e-active{
```
Comment (hein-obox) [inline modules/nested-tabs/assets/scss/frontend.scss:173]: Can we remove the `&, &.e-active`.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 2]
```diff
@@ -207,6 +207,85 @@ test.describe( 'Nested Tabs tests', () => {
...
+		// Add widgets.
+		await editor.addWidget( 'nested-tabs', container );
+		await editor.getPreviewFrame().waitForSelector( '.e-n-tab-title.e-normal.e-active' );
+
+		// Act.
+		// Set icons to tabs.
+		await setIconsToTabs( page, TabsIcons );
+
+		// Set icon color.
+		await setColor( page, editor, 'icon_section_style', 'icon_section_hover', 'icon_color_hover', '#ff0000' );
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:267]: Please call it `setIconColor`.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 3]
```diff
@@ -255,3 +334,12 @@ async function cleanup( wpAdmin ) {
 		'nested-elements': 'inactive',
 	} );
 }
+
+async function setColor( page, editor, sectionClass, tabClass, colorPickerClass, color ) {
+	await editor.activatePanelTab( 'style' );
+	await page.locator( `.elementor-control-${ sectionClass }` ).click();
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:340]: 'panelName` / 'panelClass'

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 4]
```diff
@@ -207,6 +207,85 @@ test.describe( 'Nested Tabs tests', () => {
...
+
+		// Add widgets.
+		await editor.addWidget( 'nested-tabs', container );
+		await editor.getPreviewFrame().waitForSelector( '.e-n-tab-title.e-normal.e-active' );
+
+		// Act.
+		// Set icons to tabs.
+		await setIconsToTabs( page, TabsIcons );
+
+		// Set icon color.
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:266]: '// Set icon hover color.'

File: modules/nested-tabs/assets/scss/frontend.scss [hunk 5]
```diff
@@ -181,6 +166,23 @@
 					}
 				}
 			}
+
+			&:hover {
+
```
Comment (hein-obox) [inline modules/nested-tabs/assets/scss/frontend.scss:171]: Remove empty line.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 6]
```diff
@@ -207,6 +207,85 @@ test.describe( 'Nested Tabs tests', () => {
 		await expect( activeTab ).toHaveCSS( 'margin-bottom', '50px' );
 		await expect( lastTab ).toHaveCSS( 'margin-top', '25px' );
 	} );
+
+	test( 'Check that active and non-active tabs color changes on hover', async ( { page }, testInfo ) => {
+		// Arrange.
+		const wpAdmin = new WpAdminPage( page, testInfo );
+
+		await wpAdmin.setExperiments( {
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:215]: Pull latest update from your previous task.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 7]
```diff
@@ -207,6 +207,85 @@ test.describe( 'Nested Tabs tests', () => {
...
+		const redColor = 'rgb(255, 0, 0)',
+			whiteColor = 'rgb(255, 255, 255)',
+			activeTabIcon = editor.getPreviewFrame().locator( '.e-n-tab-title.e-active > .e-n-tab-icon.e-active i' ).first(),
+			activeTabTitle = editor.getPreviewFrame().locator( '.e-n-tab-title.e-active' ).first(),
+			notActiveTabIcon = editor.getPreviewFrame().locator( '.e-n-tab-title:not(.e-active) > .e-n-tab-icon:not(.e-active) i' ).first(),
+			notActiveTabTitle = editor.getPreviewFrame().locator( '.e-n-tab-title:not(.e-active) > .e-n-tab-title-text' ).first();
+
+		// Assert.
+		// Check color differences in active tab.
+		await activeTabIcon.hover();
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:278]: Possibly first add an `.waitForSelector( `.e-n-tab-title.e-active > .e-n-tab-icon.e-active i` );`?

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 1 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss]: Done

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 2 above)
Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js]: Done

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 8]
```diff
@@ -255,3 +334,12 @@ async function cleanup( wpAdmin ) {
 		'nested-elements': 'inactive',
 	} );
 }
+
+async function setColor( page, editor, sectionClass, tabClass, colorPickerClass, color ) {
+	await editor.activatePanelTab( 'style' );
+	await page.locator( `.elementor-control-${ sectionClass }` ).click();
+	await page.locator( `.elementor-control-${ tabClass }` ).click();
```
[the remark this answers is not in the document]
Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js]: Done

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 4 above)
Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js]: Done

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 6 above)
Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:215]: Done

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 7 above)
Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:279]: Done

Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:279]: Done

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 6 above)
Comment (hein-obox) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:215]: Can you please update all of the tests to Tzvi's `.setup()` method?

**Review by hein-obox — DISMISSED** [review]

**Review by TzviRabinovitch — DISMISSED** [review]

**Review by TzviRabinovitch — APPROVED** [review]

**Pull request merged**